streams-contrib/streams-provider-twitter/src/main/java/org/apache/streams/twitter/provider/TwitterTimelineProvider.java

```diff
@@ -297,11 +294,32 @@ public void cleanUp() {
     }
 
     public void addDatum(StreamsDatum datum) {
+        if (this.lock.writeLock().tryLock()) {
+            try {
+                if (this.providerQueue instanceof BlockingQueue) {
+                    if (((BlockingQueue) this.providerQueue).remainingCapacity() > 0) {
+                        ComponentUtils.offerUntilSuccess(datum, this.providerQueue);
+                        return;
+                    } else {
+                        LOGGER.warn("Queue was at capacity, will yield and try again");
+                    }
+                } else {
+                    LOGGER.warn("Queue was not a blocking queue, will unconditionally offer data");
+                    ComponentUtils.offerUntilSuccess(datum, this.providerQueue);
+                    return;
+                }
+            } finally {
+                this.lock.writeLock().unlock();
+            }
+        } else {
+            LOGGER.warn("Lock was in use, will yield and try again");
+        }
         try {
-            lock.readLock().lock();
-            ComponentUtils.offerUntilSuccess(datum, providerQueue);
-        } finally {
-            lock.readLock().unlock();
+            Thread.sleep(5000);
+            addDatum(datum);
+        } catch (InterruptedException e) {
+            LOGGER.error("Interrupted while trying to add a datum.  Datum may be lost. {}", datum.getId());
```

Comment:
An InterruptedException is more of a control flow interrupt for threading in Java.  It would not provide additional information that would be useful, so I don't bother logging it.